galsim/wfirst/wfirst_wcs.py

```diff
@@ -0,0 +1,571 @@
+        # The other terms have been divided by the linear terms, so that these become pure
+        # multiplicative factors. There is no need to change signs of the SIP coefficents associated
+        # with odd powers of X! Change sign of a10, b10 because the tangent-plane X pixel coordinate
+        # has sign opposite to the detector pixel X coordinate, and this transformation maps pixels
+        # to tangent plane.
+        a10 = -a_sip[i_sca,1,0]
+        a11 = a_sip[i_sca,0,1]
+        b10 = -b_sip[i_sca,1,0]
+        b11 = b_sip[i_sca,0,1]
+
+        # Rotate by pa_fpa.
+        cos_pa = np.cos(pa_sca.rad())
+        sin_pa = np.sin(pa_sca.rad())
+        header['CD1_1'] = (cos_pa * a10 + sin_pa * b10, "partial of first axis coordinate w.r.t. x")
+        header['CD1_2'] = (cos_pa * a11 + sin_pa * b11, "partial of first axis coordinate w.r.t. y")
+        header['CD2_1'] = (-sin_pa * a10 + cos_pa * b10, "partial of second axis coordinate w.r.t. x")
+        header['CD2_2'] = (-sin_pa * a11 + cos_pa * b11, "partial of second axis coordinate w.r.t. y")
+        header['ORIENTAT'] = (pa_sca / galsim.degrees, 
+                              "position angle of image y axis (deg. e of n)")
+        header['LONPOLE'] = (phi_p / galsim.degrees,
+                             "Native longitude of celestial pole")
+
+        for i in range(n_sip):
+            for j in range(n_sip):
+                if i+j >= 2 and i+j < n_sip:
+                    sipstr = "A_%d_%d"%(i,j)
+                    header[sipstr] = a_sip[i_sca,i,j]
+                    sipstr = "B_%d_%d"%(i,j)
+                    header[sipstr] = b_sip[i_sca,i,j]
+
+        if not as_header:
+            wcs = galsim.FitsWCS(header=header)
+            wcs_list.append(wcs)
+        else:
+            wcs_list.append(header)
+
+    return wcs_list
+
+def findSCA(wcs_list, ra=None, dec=None, pos=None, include_border=False):
+    """
+    This is a subroutine to take a list of WCS (one per SCA) from galsim.wfirst.getWCS() and query
+    which SCA a particular real-world coordinate would be located on.  If the position is not
+    located on any of the SCAs, the result will be None.
+
+    The position should be specified in a way similar to how it is specified for
+    galsim.wfirst.getWCS().
+
+    Depending on what the user wants to do with the results, they may wish to use the
+    `include_border` keyword.  This keyword determines whether or not to include an additional
+    border corresponding to half of the gaps between SCAs.  For example, if a user is drawing a
+    single image they may wish to only know whether a given position falls onto an SCA, and if so,
+    which one (ignoring everything in the gaps).  In contrast, a user who plans to make a sequence
+    of dithered images might find it most useful to know whether the position is either on an SCA or
+    close enough that in a typical dither sequence it might appear on the SCA at some point.  Use of
```

Comment:
This is probably outside the scope of this PR, but is the WFirst survey strategy really to just use small dithers?  I thought it was pretty well accepted by just about everyone that it is a better idea to use large offsets to put objects at very different locations in the focal plane.  It vastly improves both the astrometric and photometric calibrations of the data.